Yoast/wordpress-seo
# filters unknown shortcodes

## Summary

This PR can be summarized in the following changelog entry:
Removes all shortcodes that aren't properly registered.

## Relevant technical choices:

* Apply a filter to remove everything that looks like shortcode to prevent them cluttering the text. 

Everything between `[ ]` is fltered out if it doesn't start with a number or space. 
The textstring is filtered out only if it contains a combination of alphanumeric characters and characters in the following list ` = - , . : ; _ ' "` and whitespaces. These characters are used in shortcodes (tested this with Fusion Builder, Divi, Visual Composer)

There is 1 downside to this fix. If text is places within `[ ]`, that is sometimes used to clearify quotes, they are filtered out as well, since we cannot differentiate between shortcodes and text. 

I've tested this with a couple of sitebuilders, and the impact is pretty minimal, but we do introduce something that potentially can break stuff, and I'm not really happy about it. I've also tested with another branch, which uses a similar fix: https://github.com/Yoast/wordpress-seo/blob/DT/filter-unregistered-shortcodes/js/src/wp-seo-shortcode-plugin.js#L78

What I do here is checking of it starts with something that looks like a shortcode we recognize. So starting with `vc_` in case of Visual Composer, `fusion_` in the case of fusionbuilder. The downside of this is that we cannot filter every shortcode, or shortcodes that are not prefixed (most of the plugins don't prefix their shortcodes).
The other downside is that making this list is quite extensive, and if it becomes a very large regex, it will give a performance penalty.

## Test instructions

This PR can be tested by following these steps:

* Create a post with shortcodes (easiest way is to use visual composer, that uses a bunch of [vc_ ] like shortcodes
* Look at the prominent words (in premium), these should not show the shortcodes. 
* Alternatively, look at the content of the paper, `YoastSEO.app.paper.getText()` in the console, to see the shortcodes are filtered out. 
* You should also see that adding shortcodes does not change the wordcount, or change other feedback. 

Fixes https://github.com/Yoast/wordpress-seo-premium/issues/907
Fixes https://github.com/Yoast/wordpress-seo-premium/issues/862
Fixes https://github.com/Yoast/wordpress-seo-premium/issues/854
Fixes https://github.com/Yoast/wordpress-seo-premium/issues/830
Fixes https://github.com/Yoast/wordpress-seo-premium/issues/828
Fixes #6236
Fixes #6231
Fixes #6210 
Fixes #6007 
Fixes #5983 
Fixes #5969 
Fixes #5920 
Fixes #5811 

Replaces #6272 I messed up the branches.
